src/frontend/screens/Settings/MapSettings/DeleteMapBottomSheet.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from "react";
+import BottomSheet, { BottomSheetBackdrop } from "@gorhom/bottom-sheet";
+import { BottomSheetMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import { StyleSheet, View, Text } from "react-native";
+import { MAPEO_BLUE, RED, WHITE } from "../../../lib/styles";
+import { DeleteIcon, ErrorIcon } from "../../../sharedComponents/icons";
+import Button from "../../../sharedComponents/Button";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import api from "../../../api";
+import { useNavigationFromRoot } from "../../../hooks/useNavigationWithTypes";
+import { useMapStyle } from "../../../hooks/useMapStyle";
+import { DEFAULT_MAP_ID } from "./BackgroundMaps";
+
+const m = defineMessages({
+  deleteMapMessage: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMapMessage",
+    defaultMessage: "Are you sure you want to delete {mapName}?",
+  },
+  deleteMapWarning: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMapWarning",
+    defaultMessage:
+      "This area will no longer be available offline. Cannot be undone.",
+  },
+  deleteMap: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.deleteMap",
+    defaultMessage: "Delete Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.DeleteMapBottomSheet.cancel",
+    defaultMessage: "Cancel",
+  },
+});
+
+interface DeleteMapBottomSheetProps {
+  mapName: string;
+  mapId: string;
+  closeSheet: () => void;
+}
+
+export const DeleteMapBottomSheet = React.forwardRef<
+  BottomSheetMethods,
+  DeleteMapBottomSheetProps
+>(({ mapName, closeSheet, mapId }, sheetRef) => {
+  const [snapPoints, setSnapPoints] = React.useState<(number | string)[]>([
+    0,
+    "40%",
+  ]);
+
+  const { navigate } = useNavigationFromRoot();
+  const { formatMessage: t } = useIntl();
+  const { styleId, setStyleId } = useMapStyle();
+
+  function deleteMap() {
+    // Cannot delete Default Map
+    if (mapId === DEFAULT_MAP_ID) {
+      return;
+    }
+
+    // If user is deleting the map that is currently being used, we want to set the map to be the default map
+    if (styleId === mapId) {
+      setStyleId(DEFAULT_MAP_ID);
+    }
```

Comment:
this should probably happen only if the deletion was successful i.e. `api.map.deleteStyle` resolves